### ISSUE TITLE
Internal Server Error When Requiring Players to Clear Inventory Before Joining in 1.9

### ISSUE THREAD

**Issue by Darunia18** [garbagemule/MobArena#296]
Title explains it all. When requiring an empty inventory to join an arena in 1.9, it gives an internal server error and does not allow anyone to join.

**Comment by garbagemule** [comment]
Internal server error is as descriptive as "it doesn't work", so the title really doesn't explain it all.

Duplicate.

**Comment by Darunia18** [comment]
Sorry I couldn't be more descriptive. That's the only error given. The console doesn't even give any output.

**Comment by garbagemule** [comment]
Yes, it does. It is virtually impossible to get an internal server error without an actual internal server error.

**Comment by garbagemule** [comment]
I think that came off a little more passive-aggressive than I hoped for. Sorry about that - it's been a long week.

When you see "internal server error", it means the command you entered caused an exception to be thrown, and that exception was not caught anywhere in the plugin. When this happens, there is a "default exception handler" that catches the exception and prints it to the server log/console (otherwise, the whole server would crash). The exception and its stacktrace will be available in the log if you see "internal server error".

You can verify this by stopping your server, deleting the current log file, starting your server back up, and reproducing the bug. At this point, you should be able to search for "/ma j" in the log, and you'll see the point where the command was executed, and the stacktrace should be immediately below or above that line.

In my five years of plugin development, I haven't ever experienced an internal server error that did not produce a stacktrace, and I have seen my fair share of errors and exceptions - that's why I'm so certain :)

**Comment by Darunia18** [comment]
It's ok. We've all had those weeks. Yeah I wasn't seeing anything in the console which I thought was odd, but maybe it was just something I overlooked. Later tonight I'll try again and let you know what I find.

To expand on this further: I use a host that has the console built into a webpage and it may just not be pushing every error to the web console. When I first saw the error, I believed it to be another similar "It's not working in 1.9!" error that you had said was already fixed, but was persistent on trying to join an arena without needing any update. I changed the configurations to no longer require empty inventories to join and it worked. I haven't gone back to find the error since. I just reported what I knew I saw before where I had received an internal server error message in-game, but no readings on the web console.

Also, thank you for all your hard work on this plugin. It really is greatly appreciated.
